helper/restore_helper.go

```diff
@@ -225,6 +254,7 @@ func doRestoreAgent() error {
 						logError(fmt.Sprintf("Error encountered getting restore data reader: %v", err))
 						return err
 					}
+					defer finalizeReaderPlugin(readers[contentToRestore])
```

Comment:
do we need to have a defer call here? It looks anyway we will call it for resize option:
```
			// SDF reads the pipe output by chunks in several iterations, but
			// Wait() from finalizeReaderPlugin() call will close the pipe
			// immediately. Let defer close it instead.
			if !*singleDataFile {
				finalizeReaderPlugin(readers[contentToRestore])
			}
```
But the registered defer calls will still exist. Real clusters could easily have ~10000-20000 tables.
Thus we will have the same amount of registered defer calls (wasting memory, etc), that seems we could have avoided.